fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -314,7 +315,32 @@ public List<PartitionKey> getPrunedPartitions(Table table, ScalarOperator predic
             }
 
             try {
-                partitionKeys.add(createPartitionKey(values, partitionColumns, table.getType()));
+                List<com.starrocks.catalog.Type> srTypes = new ArrayList<>();
+                for (PartitionField partitionField : spec.fields()) {
+                    if (partitionField.transform().isVoid()) {
+                        continue;
+                    }
+
+                    if (!partitionField.transform().isIdentity()) {
+                        Type sourceType = spec.schema().findType(partitionField.sourceId());
+                        Type resultType = partitionField.transform().getResultType(sourceType);
+                        if (resultType == Types.DateType.get()) {
+                            resultType = Types.IntegerType.get();
```

Comment:
why transform date type to integer

Reply:
the partition value of data type is Integer.